fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/VersionServiceImpl.java

```diff
@@ -43,17 +44,18 @@
     private VersionResourceOperationFactory versionOperationFactory;
 
     @Override
-    public void createVersion(final Transaction transaction, final String resourceId, final String userPrincipal) {
+    public void createVersion(final Transaction transaction, final FedoraId fedoraId, final String userPrincipal) {
         final var session = psManager.getSession(transaction.getId());
-        final var operation = versionOperationFactory.createBuilder(resourceId)
+        final var operation = versionOperationFactory.createBuilder(fedoraId.getResourceId())
```

Comment:
I think this might cause a problem for versioning of binary descriptions separately from the binary itself. `getResourceId` for a description returns the ID of the binary. It might be fine, but might need to be reviewed when we start passing FedoraId's into these builders.